### DISCUSSION TITLE
Glab creates new remote on `glab mr checkout -t`

### DISCUSSION THREAD

**Discussion by GordianDziwis** [profclems/glab#955] [Q&A]
The description says:

  -t, --track                    set checked out branch to track remote branch, adds remote if needed

What does glab understand under "if needed"? Because in my setup it creates a new remote.
But if I checkout a mr without the `-t` option and then setup the tracking with `git push --set-upstream foo bar` it works fine with the original repo. So I would guess adding the remote is not needed. 

Is this a bug or misunderstanding?

**Comment by zemzale** [comment]
This seems to be a pretty weird flag IMO. 

It adds a remote for each user If he doesn't have a remote. This probably only makes sense if you are checking out branches from other repos(forks usually).  The logic is really not there for automatic tracking of origin branches. 

You probably just should use the `set-upstream-to` flag.
